tools/finish_release.py

```diff
@@ -172,30 +188,39 @@ def fetch_version_number():
     # Find the build artifacts
     build_client = connection.clients.get_build_client()
     get_builds_response = build_client.get_builds('certbot', definitions='3')
-    build_id = get_builds_response.value[0].id
-    version = build_client.get_build('certbot', build_id).source_branch.split('v')[1]
-    return version
+    for build in get_builds_response.value:
+        version = build_client.get_build('certbot', build.id).source_branch.split('v')[1]
+        if major_version is None or version.split('.')[0] == major_version:
+            return version
+    raise ValueError('Release not found on Azure Pipelines!')
 
 def main(args):
     parsed_args = parse_args(args)
 
     css = parsed_args.css
-    version = fetch_version_number()
+    version = fetch_version_number('2' if parsed_args.progressive else None)
 
     # Once the GitHub release has been published, trying to publish it
     # again fails. Publishing the snaps can be done multiple times though
     # so we do that first to make it easier to run the script again later
     # if something goes wrong.        
     #
-    # For now though, we're only going to publish snaps to the stable channel
-    # for 1.x.y releases and only going to update our Windows installer for
-    # 2.x.y releases. Once we feel confident enough about Certbot 2.0, we
-    # should stop doing 1.x.y releases and unconditionally publish both snaps
+    # We only publish all snaps to the stable channel for 1.x releases. For 2.x
+    # releases, we only progressively release the base Certbot snap and update
+    # the Windows installer. Once we feel confident enough about Certbot 2.x,
+    # we should stop doing 1.x releases and unconditionally publish all snaps
     # and the Windows installer.
     if version.startswith('1.'):
-        promote_snaps(version)
+        promote_snaps(ALL_SNAPS, 'candidate', version)
+    elif not parsed_args.progressive and parsed_args.css is None:
+        # Fail fast if we weren't given a --css argument because we're going to
+        # need it later.
+        raise ValueError('Please provide the --css command line argument')
     else:
-        publish_windows(css)
+        promote_snaps(['certbot'], 'beta', version,
+                      progressive_percentage=PROGRESSIVE_RELEASE_PERCENTAGE)
```

Comment:
since 2.x snaps always trigger a progressive release, maybe we should throw an error if a 2.x version is releasing & `--progressive` wasn't specified, or just remove the option and have 2.x be implicitly progressive until we're ready to fully release?

Reply:
I think with the way things are now, 2.x is implicitly progressive (or at least that's how it's supposed to work 😳 ). What I think `--progressive` does is:

1. Finds the last 2.0 release rather than the last release of any version from Azure Pipelines.
2. Skips publishing the Windows installer.

My thought is we'll continue just running `tools/finish_release.py --css <blah>` for all releases, but we can use `--progressive` to update the deployment percentage between releases.

Reply:
Sorry I initially misunderstood your comment.

My plan here was that `tools/finish_release.py --css <blah>` would continue to automatically do The Right Thing™ depending on whether it's a 1.x or 2.x release. We'd only use `tools/finish_release.py --progressive` if we wanted to update the progressive deployment percentage without rebuilding the latest 2.x release.

Reply:
ahh i gotcha! that makes sense to me, and i think the renamed flag is clearer. thanks!